src/lvl_script_commands.c

```diff
@@ -5129,6 +5129,139 @@ static void set_player_modifier_process(struct ScriptContext* context)
+                    }
+                    break;
+                case 6: // Salary
+                    mdfradd = dungeon->modifier.pay;
+                    mdfradd = mdfradd + mdfrval;
+                    if (mdfradd >= 0) {
+                        SCRIPTDBG(7,"Adding %d to Player Modifier '%s' of player %d.", mdfrval, mdfrname, (int)plyr_idx);
+                        dungeon->modifier.pay = mdfradd;
+                    } else {
+                        SCRPTERRLOG("Unable to add %d to Player Modifier of player %d: final value of '%s' cannot be negative.", mdfrval, (int)plyr_idx, mdfrname);
+                    }
+                    break;
+                case 7: // TrainingCost
+                    mdfradd = dungeon->modifier.training_cost;
+                    mdfradd = mdfradd + mdfrval;
+                    if (mdfradd >= 0) {
+                        SCRIPTDBG(7,"Adding %d to Player Modifier '%s' of player %d.", mdfrval, mdfrname, (int)plyr_idx);
+                        dungeon->modifier.training_cost = mdfradd;
+                    } else {
+                        SCRPTERRLOG("Unable to add %d to Player Modifier of player %d: final value of '%s' cannot be negative.", mdfrval, (int)plyr_idx, mdfrname);
+                    }
+                    break;
+                case 8: // ScavengingCost
+                    mdfradd = dungeon->modifier.scavenging_cost;
+                    mdfradd = mdfradd + mdfrval;
+                    if (mdfradd >= 0) {
+                        SCRIPTDBG(7,"Adding %d to Player Modifier '%s' of player %d.", mdfrval, mdfrname, (int)plyr_idx);
+                        dungeon->modifier.scavenging_cost = mdfradd;
+                    } else {
+                        SCRPTERRLOG("Unable to add %d to Player Modifier of player %d: final value of '%s' cannot be negative.", mdfrval, (int)plyr_idx, mdfrname);
+                    }
+                break;
+                case 9: // Loyalty
+                    mdfradd = dungeon->modifier.loyalty;
+                    mdfradd = mdfradd + mdfrval;
+                    if (mdfradd >= 0) {
+                        SCRIPTDBG(7,"Adding %d to Player Modifier '%s' of player %d.", mdfrval, mdfrname, (int)plyr_idx);
+                        dungeon->modifier.loyalty = mdfradd;
+                    } else {
+                        SCRPTERRLOG("Unable to add %d to Player Modifier of player %d: final value of '%s' cannot be negative.", mdfrval, (int)plyr_idx, mdfrname);
+                    }
+                    break;
+                default:
+                    WARNMSG("Unsupported Player Modifier, command %d.", mdfrdesc);
+                    break;
+            }
+        } else
+        {
+            SCRPTERRLOG("Can't manipulate Player Modifier '%s', player %d has no dungeon.", mdfrname, (int)plyr_idx);
```

Comment:
Can't this error be done on the 'check' function instead?

Reply:
I believe it does yes. Otherwise whenever you use an 'all players' in your command you would get an error in the log right?